dvc/remote/ssh/connection.py

```diff
@@ -167,16 +167,19 @@ def upload(self, src, dest, no_progress_bar=False, progress_title=None):
         self._sftp_connect()
 
         self.makedirs(posixpath.dirname(dest))
+        tmp_file = dest + ".tmp"
```

Comment:
what if two dvc instances try to upload there at the same time? We should use a better tmp name generation here. E.g. see RemoteBase.tmp_file().